File: company/TS/1.py


def reverse_list(arr):
    left = 0
    right = len(arr)-1
    while (left < right):
        temp = arr[left]
        arr[left] = arr[right]
        arr[right] = temp
        left += 1
        right -= 1
 
    return arr

def performOperations(operations: list[int], arr: list[int]) -> None:
    for i in range(len(operations)):
        arr[operations[i][0]: operations[i][1]+1] = reverse_list(arr[operations[i][0]: operations[i][1]+1])
    

if __name__ == '__main__':
    operations = [[0, 9], [4, 5], [3, 6], [2, 7], [1, 8], [0, 9]]
    arr = [9,8,7,6,5,4,3,2,1,0]

    performOperations(operations, arr)
    print(arr)

File: company/TS/1.go
package main

import "fmt"

func performOperations(arr []int32, operations [][]int32) []int32 {
	for _, value := range operations {
		reverse(arr[value[0]: value[1]+1])
	}

	return arr
}

func reverse[S ~[]E, E any](s S) S {
	for i, j := 0, len(s)-1; i < j; i, j = i+1, j-1 {
		s[i], s[j] = s[j], s[i]
	}
	return s
}

func main() {
	operations := [][]int32{{0, 9}, {4, 5}, {3, 6}, {2, 7}, {1, 8}, {0, 9}}
	arr := []int32{9, 8, 7, 6, 5, 4, 3, 2, 1, 0}
	result := performOperations(arr, operations)
	fmt.Println(result)
}
